packages/next/src/build/index.ts

```diff
@@ -2018,13 +2020,16 @@ export default async function build(
                       pagePath
                     )
 
+                const isInsideAppDir = pageType === 'app'
                 const staticInfo = pagePath
-                  ? await getPageStaticInfo({
+                  ? await getStaticInfoIncludingLayouts({
+                      isInsideAppDir,
                       pageFilePath,
-                      nextConfig: config,
-                      // TODO: fix type mismatch
-                      pageType:
-                        pageType === 'app' ? PAGE_TYPES.APP : PAGE_TYPES.PAGES,
+                      pageExtensions: config.pageExtensions,
+                      appDir,
+                      config,
+                      isDev: false,
+                      page: originalAppPath || page,
```

Comment:
This fallback is bit unclear, we don't know when it's using `originalAppPath` or `page`.
Is originalAppPath always preseted when `pageType === app`?

Reply:
Yes!!

https://github.com/vercel/next.js/blob/f88460ebd486c325c63cc056cb7e6a33b620b995/packages/next/src/build/index.ts#L1997-L2010

https://github.com/vercel/next.js/blob/f88460ebd486c325c63cc056cb7e6a33b620b995/packages/next/src/build/index.ts#L951-L962

Reply:
Can we update it to `isApp ? originalAppPath : page`

Reply:
https://github.com/vercel/next.js/pull/68793/files#diff-90d1d5f446bdf243be25cc4ea2295a9c91508859d655e51d5ec4a3562d3a24d9R2032-R2035